arch/sim/src/Makefile

```diff
@@ -414,6 +414,9 @@ ifeq ($(CONFIG_MM_KASAN_GLOBAL),y)
 	            .kasan.global : {KEEP(*(.data..LASAN0)) KEEP (*(.data.rel.local..LASAN0)) }\n  \
 	            .interp : {*(.interp)}/g' nuttx.ld
 endif
+ifneq ($(CONFIG_SIM_CUSTOM_DATA_SECTION),"")
+	$(Q) sed -i '/\.data *:/i $(shell echo $(CONFIG_SIM_CUSTOM_DATA_SECTION))' nuttx.ld
```

Comment:
why not directly use CONFIG_SIM_CUSTOM_DATA_SECTION, but echo it?

Reply:
This config contains double quotes `"`, but the sed /i syntax does not support eliminating them.

Reply:
but should we strip quote in Makefile?

Reply:
Done